invoke/runners.py

```diff
@@ -219,6 +220,13 @@ def run(self, command, **kwargs):
             print("\033[1;37m{0}\033[0m".format(command))
         # Determine pty or no
         self.using_pty = self.should_use_pty(opts['pty'], opts['fallback'])
+        # Environment variables
+        self.env = os.environ.copy()
+        if env is not None:
+            if env:
+                self.env.update(env)
+            else:
+                self.env = {}
```

Comment:
This is a pretty big breaking change to current (and arguably expected) behaviour. If the user doesn't pass an environment, the child process is executed with no environment (empty dict). The general expected behaviour is that a child is spawned with the environment of it's parent unless otherwise specified.

This would probably be best as:

```
 # Environment variables
self.env = os.environ.copy()
if env:
    self.env.update(env)
```